src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -895,6 +895,15 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
             x.rethrowAsIOException(source);
         }
 
+        if (sourceAttrs.isDirectory()) {
+            // ensure source can be moved
+            try {
+                access(source, W_OK);
```

Comment:
I think W_OK is correct but I wonder if this should move to where ensureEmptyDir is called as that is the additional checks for directories but copying.

Reply:
That does not work. With this change the operation fails in `rename` and the existing target is deleted.